src/_P101_WakeOnLan.ino

```diff
@@ -0,0 +1,475 @@
+
+        msgStr    = F("Provided IP Invalid, Using Default. ");
+        errorStr += msgStr;
+        msgStr    = wolStr + msgStr;
+        msgStr   += F("[");
+        msgStr   += F(IP_STR_DEF_P101);
+        msgStr   += F("]");
+        addLog(LOG_LEVEL_INFO, msgStr);
+      }
+      else if (!validateIp(ipString)) { // Unexpected IP Address value. Leave as-is, but Warn User.
+        msgStr    = F("WARNING, Please Review IP Address. ");
+        errorStr += msgStr;
+        msgStr    = wolStr + msgStr;
+        msgStr   += F("[");
+        msgStr   += ipString;
+        msgStr   += F("]");
+        addLog(LOG_LEVEL_INFO, msgStr);
+      }
+
+      // Check MAC Address.
+      if (!safe_strncpy(macString, web_server.arg(getPluginCustomArgName(1)), MAC_BUFF_SIZE_P101)) {
+        // msgStr += getCustomTaskSettingsError(1); // Report string too long.
+        // errorStr += msgStr;
+        // msgStr    = wolStr + msgStr;
+        // addLog(LOG_LEVEL_INFO, msgStr);
+      }
+
+      if (strlen(macString) == 0) { // MAC Address missing, use default value.
+        strcpy_P(macString, (char *)(F(MAC_STR_DEF_P101)));
+
+        msgStr    = F("MAC Address Not Provided, Populated with Zero Values. ");
+        errorStr += msgStr;
+        addLog(LOG_LEVEL_INFO, wolStr + msgStr);
+      }
+      else if (!validateMac(macString)) { // Suspicious MAC Address. Leave as-is, but warn User.
+        msgStr    = F("ERROR, MAC Address Invalid. ");
+        errorStr += msgStr;
+        msgStr    = wolStr + msgStr;
+        msgStr   += F("[");
+        msgStr   += macString;
+        msgStr   += F("]");
+        addLog(LOG_LEVEL_INFO, msgStr);
+      }
+
+      // Save the user's IP and MAC Address parameters into Custom Settings.
+      safe_strncpy(deviceTemplate[0], ipString,   IP_BUFF_SIZE_P101);
+      safe_strncpy(deviceTemplate[1], macString, MAC_BUFF_SIZE_P101);
+
+      if (errorStr.length() > 0) { // Send error messages (if any) to webform.
+        addHtmlError(errorStr);
+      }
+
+      // Save all the Task parameters.
+      SaveCustomTaskSettings(event->TaskIndex, (byte *)&deviceTemplate, sizeof(deviceTemplate));
+      UDP_PORT_P101 = getFormItemInt(F(FORM_PORT_P101));
+      success       = true;
+      break;
+    }
+
+    case PLUGIN_INIT: {
+      success = true;
+      break;
+    }
+
+    case PLUGIN_READ: {
+      success = false;
+      break;
+    }
+
+    case PLUGIN_WRITE: {
+      bool   taskEnable  = true;
+      byte   parse_error = false;
+      String tmpString   = string;
+      String strings[2];
+
+      //  addLog(LOG_LEVEL_INFO, String(F("--> WOL taskIndex= ")) + String(event->TaskIndex)); // Debug
+
+      //  Warning, event->TaskIndex is incorrectly set in PLUGIN_WRITE if plugin is disabled.
+      //  So checking the task enable status needs special attention.
+      //  See https://github.com/letscontrolit/ESPEasy/issues/3317
+      if (event->TaskIndex < TASKS_MAX) {
```

Comment:
See `validTaskIndex` (src/Globals/Plugins.cpp)

Reply:
Ahhh. Exactly the same thing, but in a pretty function. I'll update the code using it.

Reply:
Updated file, see line 298.